File: backend/Models/Sales/Receipt.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Models.Core;

namespace backend.Models.Sales;

/// <summary>
/// Customer payment receipts for invoices or standalone receipts
/// Represents actual payments received against invoices or independent payments
/// </summary>
public class Receipt : TenantEntity
{
    /// <summary>
    /// Invoice ID - null for standalone receipts
    /// </summary>
    public int? InvoiceId { get; set; }

    /// <summary>
    /// Receipt number (sequential format: REC-YYYY-NNNN)
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string ReceiptNumber { get; set; } = string.Empty;

    [Required]
    public DateTime PaymentDate { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Amount { get; set; }

    /// <summary>
    /// Payment method (Cash, CreditCard, BankTransfer, Check, etc.)
    /// </summary>
    [Required]
    [MaxLength(50)]
    public string PaymentMethod { get; set; } = string.Empty;

    /// <summary>
    /// Reference number (check number, transaction ID, etc.)
    /// </summary>
    [MaxLength(100)]
    public string? ReferenceNumber { get; set; }

    [MaxLength(500)]
    public string? Notes { get; set; }

    /// <summary>
    /// Currency code
    /// </summary>
    [Required]
    [MaxLength(3)]
    public string Currency { get; set; } = "ILS";

    // Fields for standalone receipts (when InvoiceId is null)
    /// <summary>
    /// Customer name for standalone receipts
    /// </summary>
    [MaxLength(200)]
    public string? CustomerName { get; set; }

    /// <summary>
    /// Customer tax ID for standalone receipts
    /// </summary>
    [MaxLength(20)]
    public string? CustomerTaxId { get; set; }

    /// <summary>
    /// Description of the payment for standalone receipts
    /// </summary>
    [MaxLength(500)]
    public string? Description { get; set; }

    // Navigation properties
    public virtual Invoice Invoice { get; set; } = null!;
}

File: front/src/types/receipt.ts
// Receipt related types and interfaces

export interface CreateReceiptRequest {
  invoiceId?: number; // Optional - null for standalone receipts
  paymentDate: string; // YYYY-MM-DD format
  amount: number;
  paymentMethod: string;
  referenceNumber?: string;
  notes?: string;
  currency?: string;
  // Fields for standalone receipts (when no invoice)
  customerName?: string;
  customerTaxId?: string;
  description?: string;
}

export interface ReceiptResponse {
  id: number;
  companyId: number;
  invoiceId?: number; // Optional for standalone receipts
  receiptNumber: string;
  paymentDate: Date;
  amount: number;
  paymentMethod: string;
  referenceNumber?: string;
  notes?: string;
  currency: string;
  exchangeRate: number;
  createdAt: Date;
  updatedAt: Date;
  
  // Fields for standalone receipts
  customerName?: string;
  customerTaxId?: string;
  description?: string;
  
  // Related data
  invoice?: {
    id: number;
    invoiceNumber: string;
    totalAmount: number;
    paidAmount: number;
    customerName: string;
  };
}

export interface ReceiptListItem {
  id: number;
  receiptNumber: string;
  paymentDate: Date;
  amount: number;
  paymentMethod: string;
  referenceNumber?: string;
  currency: string;
  invoiceNumber?: string; // Optional for standalone receipts
  customerName: string;
  totalAmount?: number; // Optional for standalone receipts
  paidAmount?: number; // Optional for standalone receipts
  remainingAmount?: number; // Optional for standalone receipts
  
  // Fields for standalone receipts
  customerTaxId?: string;
  description?: string;
  isStandalone?: boolean;
}

export const PAYMENT_METHODS = [
  'מזומן',
  'כרטיס אשראי',
  'העברה בנקאית',
  'צ\'ק',
  'ביט',
  'פייפאל',
  'אחר'
] as const;

export type PaymentMethod = typeof PAYMENT_METHODS[number];
